### PR TITLE
Migrate to openapi-generator v4.2.0

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

**Comment by yue9944882** [comment]
@brendandburns @fabiokung i did the re-gen and renamed package, PTAL

**Comment by brendandburns** [comment]
@yue9944882 can you squash commits here so that there is:

1) Generated code changes
2) Additional changes

This PR is pretty hard to review in its current state.

Thanks!

**Comment by fabiokung** [comment]
@yue9944882 @brendanburns LGTM

File: kubernetes/.openapi-generator-ignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.gitignore
+.travis.yml
+*gradle*
+*sbt*
+git_push.sh
+README.md
+
+# Remove when changes in kubernetes-client/java#366 make into upstream openapi-generator
+src/main/java/io/kubernetes/client/JSON.java
```
Comment (fabiokung) [inline kubernetes/.openapi-generator-ignore]: @yue9944882 this needs to be changed now that you renamed packages

Comment (yue9944882) [reply kubernetes/.openapi-generator-ignore]: good catch

File: examples/src/main/java/io/kubernetes/client/examples/WebSocketsExample.java [hunk 2]
```diff
@@ -40,19 +40,25 @@ public static void main(String... args) throws ApiException, IOException {
         "GET",
         client,
         new WebSockets.SocketListener() {
-          public void open(String protocol, WebSocket socket) {}
+          private volatile WebSocket socket;
 
-          public void close() {
-            // Trigger shutdown of the dispatcher's executor so this process can exit cleanly.
```
Comment (brendandburns) [inline examples/src/main/java/io/kubernetes/client/examples/WebSocketsExample.java]: Why did we remove this shutdown code?

File: kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java [hunk 3]
```diff
@@ -318,7 +320,7 @@ public void write(JsonWriter out, java.sql.Date date) throws IOException {
                         if (dateFormat != null) {
                             return new java.sql.Date(dateFormat.parse(date).getTime());
                         }
-                        return new java.sql.Date(ISO8601Utils.parse(date, new ParsePosition(0)).getTime());
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java]: Why are we changing the date parser?

File: util/src/main/java/io/kubernetes/client/Exec.java [hunk 4]
```diff
@@ -280,7 +283,7 @@ protected void handleMessage(int stream, InputStream inStream) throws IOExceptio
             }
 
             @Override
-            public void failure(Exception ex) {
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/Exec.java]: Why is this changing?

File: util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java [hunk 5]
```diff
@@ -94,27 +93,22 @@ protected void handleMessage(int stream, InputStream inStream) throws IOExceptio
...
   public synchronized void close() {
     if (state != State.CLOSED) {
       state = State.CLOSED;
-      try {
-        if (null != socket) {
-          // code 1000 means "Normal Closure"
-          socket.close(1000, "Triggered client-side terminate");
-          log.debug("Successfully closed socket.");
-        }
-      } catch (IOException ex) {
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java]: why are we removing this catch?

**Comment by brendandburns** [comment]
Some small comments.

**Comment by fabiokung** [comment]
@brendandburns I believe most (if not all) of these are related to changes from OkHttp2 -> OkHttp3

File: examples/src/main/java/io/kubernetes/client/examples/WebSocketsExample.java (hunk 2 above)
Comment (yue9944882) [reply examples/src/main/java/io/kubernetes/client/examples/WebSocketsExample.java]: it's the interface changed so we don't need this method anymore

File: kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java (hunk 3 above)
Comment (yue9944882) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java]: it's a re-play of https://github.com/kubernetes-client/java/pull/366

File: util/src/main/java/io/kubernetes/client/Exec.java (hunk 4 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/Exec.java]: interface change, or the code doesn't compile

File: util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java (hunk 5 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java]: the method signature changed, it doesn't throw IOException now

**Comment by brendandburns** [comment]
Ok, this looks reasonable. Are we ready to merge this change?

**Comment by yue9944882** [comment]
yeah, let's get it in

**Pull request merged**